integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/RegistryResourcesTestCase.java

```diff
@@ -0,0 +1,1450 @@
+import org.wso2.esb.integration.common.utils.common.TestConfigurationProvider;
+
+public class RegistryResourcesTestCase extends ESBIntegrationTest {
+
+    private String accessToken;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+    }
+
+    @Test(groups = { "wso2.esb" }, priority = 2, description = "Test adding registry content (text)")
+    public void testRegistryPostText() throws IOException {
+
+        if (!isManagementApiAvailable) {
+            Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(DEFAULT_TIMEOUT, TimeUnit.SECONDS)
+                    .until(isManagementApiAvailable());
+        }
+
+        String endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + portOffset) + "/management/"
+                + "registry-resources/content";
+        String registryPath = "registry/config/testFolder/test-text.txt";
+        String queryParameters = "?path=" + registryPath
+                + "&mediaType=testMediaType";
+        File file = new File(TestConfigurationProvider.getResourceLocation(ESBTestConstant.ESB_PRODUCT_GROUP)
+                + "/registry-resources/test-initial.txt");
+
+        SimpleHttpClient client = new SimpleHttpClient();
+        HttpResponse response = client.doPostWithMultipart(endpoint + queryParameters, file, getHeaderMap());
+        String responsePayload = client.getResponsePayload(response);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+        JSONObject jsonResponse = new JSONObject(responsePayload);
+        Assert.assertEquals(jsonResponse.get("message"), "Successfully added the registry resource");
+    }
+
+    @Test(groups = { "wso2.esb" }, priority = 2, description = "Test adding registry content (text) for test delete")
+    public void testRegistryPostTextExtraTextFile() throws IOException {
+
+        if (!isManagementApiAvailable) {
+            Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(DEFAULT_TIMEOUT, TimeUnit.SECONDS)
+                    .until(isManagementApiAvailable());
+        }
+
+        String endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + portOffset) + "/management/"
+                + "registry-resources/content";
+        String registryPath = "registry/config/testFolder/test-text-delete.txt";
+        String queryParameters = "?path=" + registryPath
+                + "&mediaType=testMediaType";
+        File file = new File(TestConfigurationProvider.getResourceLocation(ESBTestConstant.ESB_PRODUCT_GROUP)
+                + "/registry-resources/test-initial.txt");
+
+        SimpleHttpClient client = new SimpleHttpClient();
+        HttpResponse response = client.doPostWithMultipart(endpoint + queryParameters, file, getHeaderMap());
+        String responsePayload = client.getResponsePayload(response);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
```

Comment:
Add a message to be displayed when the assert fails

Reply:
Added